app/statistics/statistics.js

```diff
@@ -61,4 +61,17 @@ app.controller('StatisticsCtrl', function($scope, $routeParams, searchService, t
         loadStatistics();
     });
 
+    /**
+     * process request and start download
+     */
+    $scope.submit = function() {
+        $scope.downloadPromise = downloadService.getStatisticsData('application/vnd.ms-excel', $routeParams, "blob");
+        $scope.downloadPromise.then(function(response) {
+            var now = new Date();
+            var filename = 'QuickGO-statistics-' + now.getTime() + '-' + now.toISOString().split('T')[0].replace(/-/g,'')
+                + '.xls';
+            saveAs(response.data, filename);
+        });
```

Comment:
Not sure we do it everywhere yet, but would be nice to output something to the console if the promise fails.